### PR TITLE
Enable shared library build; create opencl.h header file; add clEnqueueFillBuffer

### PR DESCRIPTION
**Pull request by hughperkins**
Did the following changes, to enable use with clBLAS, on both linux and Windows:
- enabled shared library build, using option BUILD_SHARED_LIBRARY (default OFF, ie static, as before)
- added option to bulid tests (default ON, ie build clewTest, as before)
- added readme, with description, and install/build instructoins
- added 1.2 api function clEnqueueFillBuffer, which is used by clBLAS
- created opencl include files CL/cl.h and CL/opencl.h, which simply include clew.h, and which we can provdie to other libraries, eg clBLAS, so they build ok, without errors

Tested on:
- Ubuntu 14.04, using both shared and unshared library build
- Windows 7, using Visual Studio 2010, using both shared and unshared library build

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +11,13 @@ endif()
 include(CMakeParseArguments)
 
 
+if( "${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" )
+  set( ON_LINUX 1 )
+endif()
+if( "${CMAKE_SYSTEM_NAME}" STREQUAL "Windows" )
```
Comment (martijnberger) [inline CMakeLists.txt]: What is wrong with if( UNIX AND NOT APPLE) and if(WIN32)  ?

**Comment by martijnberger** [comment]
Thanks a lot for this. Ill merge this asap pending the conversation about the platform detection in cmake stuff.

**Comment by hughperkins** [comment]
Updated :-)

**Comment by hughperkins** [comment]
Thanks! :-)

**Pull request merged**